### PR TITLE
Multiple certs in JKS store

### PR DESCRIPTION
**Pull request by spark404**
I'm using java-apns in an environment the sends push notifications for multiple applications that all require difference certificates. Instead of managing multiple p12 files i've combined them all in a single JKS file and i'm using the aliases to determine the certificate used. This pull request moves that feature to java-apns.

I've also made some other small changes; 
* If the SSL handshake fails (due to invalid certificate) for example the client shouldn't retry as thats a fatal error
* The test stub server didn't actually use client certificates. To better simulate and test  actual certificate handling i made the stub server require certificates
* The test certificates were expired so i regenerated them and added a script to do so again if needed.
* refactored the newSSLContext bits into a bulder

Hope this is a useful addition to java-apns

### CODE REVIEW

**Comment by spark404** [comment]
I think the issue with Travis is timeout related. "Works-on-my-laptop"(tm)

Apache Maven 3.0.5 (r01de14724cdef164cd33c7c8c2fe155faf9602da; 2013-02-19 14:51:28+0100)
Maven home: /opt/apache-maven-3.0.5
Java version: 1.6.0_65, vendor: Apple Inc.
Java home: /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home
Default locale: en_US, platform encoding: MacRoman
OS name: "mac os x", version: "10.10.5", arch: "x86_64", family: "mac"
...
Results :

Tests run: 111, Failures: 0, Errors: 0, Skipped: 3

**Comment by tomb50** [comment]
This looks awesome, can't wait to be able to use it - we have a single jks keystore and would like to be able to use it to store multiple certs for various iOS apps

**Comment by matzew** [comment]
re-triggered the tests

File: src/main/java/com/notnoop/apns/ApnsServiceBuilder.java [hunk 1]
```diff
@@ -45,14 +57,15 @@
...
-import com.notnoop.apns.internal.*;
-import com.notnoop.exceptions.InvalidSSLConfig;
-import com.notnoop.exceptions.RuntimeIOException;
-
-import static com.notnoop.apns.internal.Utilities.*;
+import static com.notnoop.apns.internal.Utilities.PRODUCTION_FEEDBACK_HOST;
+import static com.notnoop.apns.internal.Utilities.PRODUCTION_FEEDBACK_PORT;
+import static com.notnoop.apns.internal.Utilities.PRODUCTION_GATEWAY_HOST;
+import static com.notnoop.apns.internal.Utilities.PRODUCTION_GATEWAY_PORT;
+import static com.notnoop.apns.internal.Utilities.SANDBOX_FEEDBACK_HOST;
```
Comment (matzew) [inline src/main/java/com/notnoop/apns/ApnsServiceBuilder.java]: :+1:

File: src/main/java/com/notnoop/apns/internal/SSLContextBuilder.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
+package com.notnoop.apns.internal;
```
Comment (matzew) [inline src/main/java/com/notnoop/apns/internal/SSLContextBuilder.java]: mind adding a license header here ? 

I think you can skipt this line in the header:
https://github.com/spark404/java-apns/blob/multiple-certs-in-store/src/main/java/com/notnoop/apns/ApnsServiceBuilder.java#L2

(or use your own name)

Comment (spark404) [inline src/main/java/com/notnoop/apns/internal/SSLContextBuilder.java]: Fixed. I noticed some more files with license issues, so i added the mycila license check plugin. I had to reformat the licenses a bit to get the plugin to work with the default settings, but the licenses itself are not changed. By default you get the license with your name in it, but you can opt to remove the copyright header for files with multiple authors and it won't complain.

**Comment by matzew** [comment]
@spark404 landed :tada:

**Pull request merged**